### PR TITLE
[CSM Portal] Add change request approval action and problem creation

### PR DESCRIPTION
**Pull request by rksk**
Change requests had no path for a regular user to move a request out of its initial state — only a raw, unchecked field-level patch existed, and the existing approve/reject actions only appear once a request already reaches a later stage. Separately, problem records could only be searched and viewed — there was no way to create one anywhere in the stack. This PR adds both capabilities, propagated through the entity service, the CSM portal backend, and the webapp.

- Let a regular user move a change request out of its initial state via a single, backend-validated action, and let the UI discover when that action is legal from the change request's own detail response (no hardcoded state checks in the frontend).
- Add a create endpoint and a create form for problems, consistent with how change requests and incidents are already created.

**Change request approval action**
- Entity service: `PatchChangeRequestRequest` gains `RequestApproval *bool`; the change request detail type gains `LegalNextStates []string`. Both are pure passthrough — the data source performs all transition validation.
- Backend: this service is a byte-level passthrough proxy for change requests, so the new field and the new response field flow through automatically; the work here is documenting both in `openapi.yaml` and adding direct handler test coverage for the patch endpoint (which had none before).
- Webapp: a data-driven "Request approval" button on the change request detail page, shown only when `legalNextStates` includes `"assess"` — mirrors the existing `CaseActionBar` pattern of rendering off a backend-provided list rather than a hardcoded from-state check. Uses the existing patch mutation hook, which already invalidates the detail query on success, so the page updates without a manual reload. Errors surface through the existing error-banner pattern.

**Problem creation**
- Entity service: new `CreateProblem` method on the problem service, new request domain type, reusing the existing problem-detail type for the created-record response. Origin-case and primary-incident IDs are converted between the public UUID and the data source's native id the same way every other cross-link in this file is.
- Backend: new `POST /problems` handler mirroring the existing problem read handlers' auth pattern (401 with no authenticated user, no per-record access check since problems aren't project-scoped), a new entity-service client method, route registration, and `openapi.yaml` coverage including all four standard error responses.
- Webapp: a new create-problem page (subject required; category/subcategory/origin-case-id/primary-incident-id optional), a new API hook, a "New problem" entry point on the problems tab (which had none before), and route registration — all mirroring the existing change-request/incident create flows. No priority field: priority isn't settable on creation for this record type.

- As a CS engineer, I can move a change request out of its initial state without an unchecked, admin-only field edit.
- As a CS engineer, I can create a new problem record directly from the portal instead of only viewing existing ones.

Added a "Request approval" action for change requests and a problem-creation flow, including entity-service, backend, and webapp support.

## Documentation
N/A — internal CS-engineer portal feature, no external product documentation is maintained for this workflow.

## Training
N/A — no training content is maintained for this internal portal.

## Certification
N/A — no certification exam covers this internal portal.

## Marketing
N/A — internal-only feature, not customer facing.

## Automation tests
 - Unit tests
   > Added/extended Go handler tests and mocks in the CSM portal backend for both the change request patch endpoint (previously untested directly) and the new problem-creation handler (success and unauthenticated cases). Added webapp component/hook tests for the new problem-creation page and the change request detail page's new action button.
 - Integration tests
   > None added at this layer; the underlying data-source behavior was already live-verified end to end (state transition + resulting downstream automation, and record creation with cross-links) before this PR.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? N/A — this is a Go/TypeScript codebase, not Java; `go vet` and `eslint` were run clean instead
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A — no sample projects reference this API.

## Related PRs
None.

## Migrations (if applicable)
N/A — no schema or data migration involved.

## Test environment
Verified locally: `go build`/`go test`/`go vet` (entity service), `make build`/`make test`/`make vet` (CSM portal backend), `pnpm build`/`pnpm test`/`pnpm lint` (webapp).

Followed this codebase's existing conventions throughout: the sibling boolean action-flag fields already on the change request patch type, the existing case-detail action-bar pattern for backend-driven legal-transition buttons, and the existing change-request/incident create-page structure for the new problem create page.

## Summary by CodeRabbit

* **New Features**
  * Added ServiceNow “Create problem” support (POST /problems) with a new guided page at `/operations/problems/new`.
  * Expanded the problems list to show State, Assignment group, and Assigned to.
  * Added type-ahead selectors for Origin case and Primary incident during problem creation.
  * Added “Request approval” for eligible change requests based on modeled legal next states.
* **Bug Fixes**
  * Improved validation and error responses for problem creation and change-request approval flows, with updated tests.

### CODE REVIEW

**Comment by rksk** [comment]
@coderabbitai resume

**Comment by rksk** [comment]
@coderabbitai resume

**Comment by rksk** [comment]
@coderabbitai resume

File: apps/csm-portal/backend/internal/handler/problems.go [hunk 1]
```diff
@@ -79,6 +80,41 @@ func (h *ProblemHandler) SearchProblems(w http.ResponseWriter, r *http.Request)
...
+		writeError(w, http.StatusBadRequest, errMsgReadBody)
+		return
+	}
+
+	if !json.Valid(body) {
+		writeError(w, http.StatusBadRequest, ErrMsgBadRequest)
+		return
+	}
+
+	result, err := h.entity.CreateProblem(r.Context(), body)
```
[the remark this answers is not in the document]
Comment (rksk) [reply apps/csm-portal/backend/internal/handler/problems.go:152]: Fixed in 480d149a1: `CreateProblem` now decodes into a strict typed struct (`json.Decoder` with `DisallowUnknownFields`), rejects trailing/non-object JSON, requires a non-blank `subject`, and validates the optional `originCaseId`/`primaryIncidentId` UUID fields — mirroring the existing `validateCreateIncidentBody` pattern in incidents.go. Only the validated raw body is still forwarded to the entity service.

File: apps/csm-portal/backend/openapi.yaml [hunk 2]
```diff
@@ -3436,6 +3487,22 @@ components:
...
+          type: boolean
+          description: >-
+            Records the customer's post-implementation review decision.
+            Mutually exclusive with isCustomerApproved and requestApproval.
+        requestApproval:
+          type: boolean
+          description: >-
+            Transitions the change request from New to Assess and requests
+            approval. Mutually exclusive with isCustomerApproved and
+            isCustomerReviewed.
```
[the remark this answers is not in the document]
Comment (rksk) [reply apps/csm-portal/backend/openapi.yaml:3513]: Fixed in 919c0a3c9: `PatchChangeRequest` in entity-service now rejects the request with a ValidationError when more than one of `isCustomerApproved`, `isCustomerReviewed`, and `requestApproval` is set. Also added an `allOf`/`not` constraint on `PatchChangeRequestPayload` in openapi.yaml so the mutual exclusivity is enforced in the schema, not just described in the field docs.

File: entity-service/internal/service/interfaces.go [hunk 3]
```diff
@@ -375,6 +375,10 @@ type ProblemService interface {
 	// GetProblem returns the full detail of a single problem by its UUID.
 	// A NotFoundError is returned if the problem does not exist.
 	GetProblem(ctx context.Context, id string) (domain.ProblemDetail, error)
+
+	// CreateProblem creates a new problem. Subject and OriginCaseID are required.
+	// Supported by the ServiceNow data source only.
+	CreateProblem(ctx context.Context, req domain.CreateProblemRequest) (domain.ProblemDetail, error)
```
[the remark this answers is not in the document]
Comment (rksk) [reply entity-service/internal/service/interfaces.go:381]: Fixed in 0db5f0087: updated the ProblemService.CreateProblem doc comment to say only Subject is required and OriginCaseID is optional. Doc-only change, no validation logic added or altered.

File: entity-service/internal/service/sn_problem_service.go [hunk 4]
```diff
@@ -160,6 +170,70 @@ func (s *snProblemService) GetProblem(ctx context.Context, id string) (domain.Pr
...
+		return domain.ProblemDetail{}, err
+	}
+
+	var resp snCreateProblemResponse
+	if err := json.Unmarshal(raw, &resp); err != nil {
+		return domain.ProblemDetail{}, fmt.Errorf("sn create problem: parse response: %w", err)
+	}
+
+	return mapSNProblemDetailToView(resp.Problem), nil
+}
```
[the remark this answers is not in the document]
Comment (rksk) [reply entity-service/internal/service/sn_problem_service.go:231]: Fixed in 9e754011e: CreateProblem now rejects a blank/whitespace-only Subject with an apierror.ValidationError before calling out to the backing data source, matching the existing style used in PatchChangeRequest for Title.

**Comment by rksk** [comment]
@coderabbitai resume

**Review by cloby99 — APPROVED** [review]

File: apps/csm-portal/backend/internal/handler/problems.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rksk) [reply apps/csm-portal/backend/internal/handler/problems.go:152]: Already fixed on main in 480d149a1 (committed just before this PR merged) — CreateProblem now decodes into a strict typed struct via validateCreateProblemBody: unknown fields rejected, subject required and non-blank, and originCaseId/primaryIncidentId validated as UUIDs when present.

File: apps/csm-portal/backend/openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rksk) [reply apps/csm-portal/backend/openapi.yaml:3513]: Already fixed on main in 919c0a3c9 (committed just before this PR merged) — added allOf/not constraints to the PatchChangeRequest schema enforcing the mutual exclusivity, plus a matching check in sn_change_request_service.go's service layer.

File: entity-service/internal/service/interfaces.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rksk) [reply entity-service/internal/service/interfaces.go:381]: Already fixed on main in 0db5f0087 (committed just before this PR merged) — the doc comment now correctly says OriginCaseID is optional.

File: entity-service/internal/service/sn_problem_service.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rksk) [reply entity-service/internal/service/sn_problem_service.go:231]: Already fixed on main in 9e754011e (committed just before this PR merged) — CreateProblem now rejects a blank/whitespace-only Subject before forwarding to ServiceNow.
